alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -396,17 +406,30 @@ txsize :: Tx era -> Integer
 txsize (TxConstr (Memo _ bytes)) = fromIntegral (SBS.length bytes)
 
 minfee ::
-  PParams era ->
+  ( HasField "_minfeeA" (Core.PParams era) Natural,
+    HasField "_minfeeB" (Core.PParams era) Natural,
+    HasField "_prices" (Core.PParams era) Prices
+  ) =>
+  Core.PParams era ->
   Tx era ->
   Coin
 minfee pp tx =
   (txsize tx <×> a pp)
     <+> b pp
-    <+> (scriptfee (_prices pp) totExunits)
+    <+> (scriptfee (getField @"_prices" pp) allExunits)
+  where
+    a protparam = Coin (fromIntegral (getField @"_minfeeA" protparam))
+    b protparam = Coin (fromIntegral (getField @"_minfeeB" protparam))
+    allExunits = totExunits tx
+
+-- The only thing that keeps minfee from working on Core.Tx is that
+-- not all eras can extract a ExUninits from a Core.Tx. For the Alonzo
+-- era, we use this function, specialized to the Alonzo Tx defined in this file.
+-- If we had instances (HasField "exUnits" (Core.Tx era) ExUnits) we'd be golden
```

Comment:
I think there's a relevant point here - this function shouldn't be entirely general, since the way we calculate the minimum fee has changed between eras. The type of this function helps clarify that. But it could be made more general by lifting the `HasField "exUnits" (Core.Tx era) ExUnits` to a constraint on the function, which would allow the Tx to vary provided it still contains `ExUnits`.

Reply:
I did this before I read your comment. I am learning to think like you.